### ISSUE TITLE
Distinction between "Play Area" and "Control Panel"

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/john-travoltage#200]
@terracoda @emily-phet in the keyboard navigation meetings we have been having with the design team, we have been distinguishing between the "Play Area" and the "Control Panel" for consistency.

This sim doesn't have that distinction, at least not literally in the DOM.
![capture](https://cloud.githubusercontent.com/assets/6396244/23763988/155580e6-04b9-11e7-92d9-d7687e7dc4e9.PNG)

We could put the sliders in a section labelled with "Play Area" and put the other buttons in a section labelled by "Control Panel".

@emily-phet @terracoda is this something that should be done in this sim?

**Comment by terracoda** [comment]
@jessegreenberg, I would also put the sliders in the _Play Area_, and the other buttons in the _Control Panel_.

Also, out of curiosity, are you making a visualized PDOM for this sim?

**Comment by jessegreenberg** [comment]
Ok, I will add them.

@terracoda yes there is a visualized PDOM for john-travoltage, there is a build task now that can quickly generate it for any sim:

http://www.colorado.edu/physics/phet/dev/html/john-travoltage/1.3.0-dev.14/john-travoltage-a11y-view.html

**Comment by jessegreenberg** [comment]
At the moment, all interactive components are contained inside a form. Is it OK if we replace the form with these sections?

**Comment by jessegreenberg** [comment]
@jhung or @jobara can you please OK the change as well since it involves changing the HTML slightly?

http://www.colorado.edu/physics/phet/dev/html/john-travoltage/1.3.0-dev.15/john-travoltage_en.html

The change can be summarized with:
- Remove the "form" element from the DOM
- Move the appendage sliders into a section, aria-labelledby an H2 with  "Play Area" as the content, and aria-describedby the scene summary
- Move the Reset All and Mute Sound button into a section, aria-labelledby an H2 with "Control Panel" as the content.

Does this seem OK?

**Comment by jobara** [comment]
@jessegreenberg I think switching to sections is fine. However I was looking over the implementation in the example you linked to and think there are a couple of improvements that could be made.

* Move the `<h2>`'s into the sections, and remove the aria-labelledby
* Reduce the number of `<div>`'s used

**Comment by jessegreenberg** [comment]
Thanks for reviewing @jobara.  h2's were moved into the sections in the above commit and I removed the aria-labelledby's.

Is the number of ```<div>```'s acceptable in this version?
http://www.colorado.edu/physics/phet/dev/html/john-travoltage/1.3.0-dev.16/john-travoltage_en.html

Here is the accessible HTML:

```html
<article id="parent-container-60-121">
    <h1 id="label-60-121">‪John Travoltage‬</h1>
    <p id="description-60-121">John's hand is close to the doorknob, and he is ready to swing his leg to rub his foot on the rug.</p>
    <div id="60-121">
        <section id="parent-container-60-121-130">
            <h2 id="label-60-121-130">Play Area</h2>
            <div id="60-121-130" aria-describedby="description-60-121">
                <div id="parent-container-60-121-130-132">
                    <input type="range" tabindex="0" role="slider" min="0" max="30" step="1" aria-controls="polite-status" aria-valuetext="Position 17: foot rubbing on the rug" id="60-121-130-132">
                    <label id="label-60-121-130-132" for="60-121-130-132"></label>
                </div>
                <div id="parent-container-60-121-130-136">
                    <input type="range" tabindex="0" role="slider" min="0" max="100" step="1" aria-controls="polite-status" aria-valuetext="Position 64: close to the doorknob" id="60-121-130-136">
                    <label id="label-60-121-130-136" for="60-121-130-136"></label>
                </div>
            </div>
        </section>
        <section id="parent-container-60-121-131">
            <h2 id="label-60-121-131">Control Panel</h2>
            <div id="60-121-131">
                <div id="parent-container-60-121-131-157-148">
                    <input type="button" aria-pressed="false" id="60-121-131-157-148">
                    <label id="label-60-121-131-157-148" for="60-121-131-157-148">Mute Sound</label>
                </div>
                <input type="button" aria-label="‪Reset All‬" id="60-121-131-157-154">
            </div>
        </section>
    </div>
</article>
```

Some ```div```'s are necessary to structure elements so that Scenery knows where to place node's with more than a single DOM element.

**Comment by jessegreenberg** [comment]
Labels were missing on the range inputs (noticed in the HTML above), fixed in https://github.com/phetsims/john-travoltage/commit/99c8aa161e0ed7863cf24c223d0b7e2acc23d16f

**Comment by jobara** [comment]
@jessegreenberg is it possible to write these like below or are the `<div>`'s needed?

```html
<section id="parent-container-60-121-130"  aria-describedby="description-60-121">
    <h2 id="label-60-121-130">Play Area</h2>
    <input type="range" tabindex="0" role="slider" min="0" max="30" step="1" aria-controls="polite-status" aria-valuetext="Position 17: foot rubbing on the rug" id="60-121-130-132">
    <label id="label-60-121-130-132" for="60-121-130-132"></label>
    <input type="range" tabindex="0" role="slider" min="0" max="100" step="1" aria-controls="polite-status" aria-valuetext="Position 64: close to the doorknob" id="60-121-130-136">
    <label id="label-60-121-130-136" for="60-121-130-136"></label>
</section>
```

**Comment by terracoda** [comment]
@jessegreenberg, I would agree with @jobara. We should strive to reduce the number of divs (if possible).

**Comment by terracoda** [comment]
@jessegreenberg, for future reference, 

> section elements should contain an identifying element, typically a heading.

Link to [specification on section](https://www.w3.org/TR/html5/sections.html#the-section-element).

**Comment by jessegreenberg** [comment]
@jobara @terracoda with the current implementation, those ```<div>```'s are necessary. The parallel DOM is built off of the structure of Scenery's scene graph.  Say our scene graph looks like

- play area
  - body
    - appendage
    - appendage

and the appendages must look like
```html```
<input ...>
<label>...</label>
```

Since one Scenery node (an appendage) needs two HTML elements, containing both in a div makes it easy for scenery to place both of these elements in the correct place in the document.

The div under the ```h2``` section element is similarly used to make sure that the <h2> label is ordered before all other content under the ```<section>```.

Is this number of divs problematic for accessibility or HTML content?

**Comment by jessegreenberg** [comment]
https://github.com/phetsims/john-travoltage/issues/200#issuecomment-286466937

Thanks @terracoda, good to know!

**Comment by terracoda** [comment]
@jessegreenberg 
> Is this number of divs problematic for accessibility or HTML content?

This is a good question. `div`'s are non-semantic, so they mainly cause accessibility trouble if they are being used instead of a more semantically appropriate element. In the sim implementation, the `div`s are being used for non-semantic purposes, so this is good. I'm not sure how or if the `div`s affect Accessibility Tree.

To keep HTML code clean (easy to read and debug) and to keep page weight low, I always strive for as few `div`s as possible. HTML5 has so many semantically rich containers that this has become much easier in typical website development.

I haven't noticed any accessibility issues presented by the `div`s in recent interviews. Users are able to find headings and interactive elements easily.

One thing we discussed early on for _Balloons and Static Electricity_ was wrapping the controls in their labels, but this technique, though valid, has presented issues for AT users in the past ([Faulkner, 2011](https://www.paciellogroup.com/blog/2011/07/html5-accessibility-chops-form-control-labeling/)).

@jobara, do you know if `div`s used for non-semantic purposes affect the Accessibility Tree negatively?

**Comment by jobara** [comment]
@terracoda and @jessegreenberg I'm not aware of any a11y concerns when using `<div>` for none semantic purposes. That is sort of what they are for, generic containers, typically for styling / presentation needs. It's good practice to keep the markup and DOM as concise as possible, for readability, maintainability, and etc. Probably not a huge issue here.

**Comment by jessegreenberg** [comment]
Understood, thanks @terracoda and @jobara.  In this case we are trading a bit of readability/maintainability in the DOM for enhanced readability/maintainability in the sim JavaScript.

**Comment by jessegreenberg** [comment]
Thanks for the review everyone, I think this issue can be closed.
